app/models/user.rb

```diff
@@ -21,8 +21,7 @@ def exercise_status_hash
   def generate_exercise_status_hash
     result = Hash.new
     exercise_statuses.each do |exercise_status|
-      status = exercise_status.finished ? "is-finished" : "is-incomplete"
-      result[exercise_status.exercise_id] = status
+      result[exercise_status.exercise_id] = exercise_status.status
     end
     return result
```

Comment:
I would take a look at the [Enumerable](http://ruby-doc.org/core-2.2.1/Enumerable.html) mixin to see if you can write this in a more efficient manner. There are a lot of methods there that start with an initial collection and return the new result once it's filled up automatically. Take a look at the `reduce` method or see if there's something more appropriate for your need.